test/config/utility.cc

```diff
@@ -937,6 +937,18 @@ void ConfigHelper::setBufferLimits(uint32_t upstream_buffer_limit,
   RELEASE_ASSERT(bootstrap_.mutable_static_resources()->listeners_size() == 1, "");
   auto* listener = bootstrap_.mutable_static_resources()->mutable_listeners(0);
   listener->mutable_per_connection_buffer_limit_bytes()->set_value(downstream_buffer_limit);
+  const uint32_t stream_buffer_size = std::max(
+      downstream_buffer_limit, Http2::Utility::OptionsLimits::MIN_INITIAL_STREAM_WINDOW_SIZE);
```

Comment:
This may result in config being set to a value that is too large for QUIC.  I think you also need min(quic_stream_max, max(...))